tools/AzDev/src/Cmdlets/Swagger/OpenSwaggerCmdlet.cs

```diff
@@ -0,0 +1,62 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using AzDev.Models.Inventory;
+using AzDev.Services;
+
+namespace AzDev.Cmdlets.Swagger
+{
+    /// <summary>
+    /// Open-DevSwagger -Search "Graph"
+    /// Get-AzDevModule -Name "Graph" | Get-AzDevProject | Get-AzDevSwagger | Open-DevSwagger ?
+    /// </summary>
+    [Cmdlet("Open", "DevSwagger")]
+    public class OpenSwaggerCmdlet : DevCmdletBase
+    {
+        public const string SearchParameterSet = "Search";
+
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = SearchParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string Search { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            base.ProcessRecord();
+            Codebase codebase = AzDevModule.GetComponent<ICodebaseProvider>(nameof(ICodebaseProvider)).GetCodebase()
+                ?? throw new PSInvalidOperationException("Codebase is not loaded. Please run Set-DevContext first.");
```

Comment:
should unify the exception, filesnotfound was thrown from parent class

Reply:
Makes sense. This was written before the parent class introduced error handling. Will revise.